### PR TITLE
fix: `ids` prop is not `Partial`

### PR DESCRIPTION
**Pull request by abdel-17**
To override ids on some components, you have to provide ids for all elements. This PR fixes that by making the `ids` prop `Partial`.

### CODE REVIEW

File: src/lib/builders/tooltip/types.ts [hunk 1]
```diff
@@ -32,7 +32,7 @@ export type CreateTooltipProps = {
 	/**
 	 * Optionally override the default ids we assign to the elements
 	 */
-	ids?: IdObj<TooltipIdParts>;
+	ids?: Partial<IdObj<TooltipIdParts>>;
```
Comment (TGlide) [inline src/lib/builders/tooltip/types.ts:35]: You can make the IdObj itself be a partial

Comment (abdel-17) [reply src/lib/builders/tooltip/types.ts:35]: Tried it. The entire project exploded with type errors everywhere. I tried to fix them, but the calendar builder relies on the ids being required, not partial.

Comment (abdel-17) [reply src/lib/builders/tooltip/types.ts:35]: You’ll find the original commit in the history. Try it for yourself.

Comment (huntabyte) [reply src/lib/builders/tooltip/types.ts:35]: Yeah @TGlide the issue with making the `IdObj` a partial is that the type is also used internally within the builders for the `ids` object so it expects all the properties to exist.

Comment (TGlide) [reply src/lib/builders/tooltip/types.ts:35]: Yeah, it's probbly not worth the hassle and confusion. Merging!

**Review by TGlide — APPROVED** [review]

**Pull request merged**